policies/templates/gcp_enforce_labels_v1.yaml

```diff
@@ -26,79 +26,106 @@ spec:
         openAPIV3Schema:
           properties: 
             mandatory_labels:
-              description: "List of mandatory labels to check for on all supported resources. Each missing label will result in a violation."
+              description: | 
+                "List of mandatory label objects to check for on all supported resources. 
+                Label objects should be as follow:
+                label_key: label_value_regex_to_match 
+                Each missing label will result in a violation for scanned resources.
+                Each mismatching label will result in a violation for scanned resources.
+                "
+              type: array
+              items: objects
+            resource_types_to_scan:
+              description: | 
```

Comment:
instead of specifying whether something is optional in the comment, have we considered using the `required` property to denote the required fields? https://tools.ietf.org/html/draft-wright-json-schema-validation-00#section-5.15

Reply:
I think that would help clarify the usage here, if it's possible. I have not seen it used in the other constraints, but they may not have optional parameters...

Reply:
I am not sure how to implement that the right way, is it a simple

`required: true` as part of mandatory_labels (and the opposite for the optional parameter)?

Reply:
My understanding is that `required` field take a list of field names; it's at the `properties` level. 
So it'd look something like
```
properties:
  mandatory_labels:
   ...
required: ["mandatory_labels"]
```

Also while we're here, should we also define the schema for the objects in `mandatory_labels`

Reply:
I updated the required property as you mentioned, that makes sense. 

I am not sure how to define the objects in mandatory_labels. Do you have an example I can reuse?

Reply:
unfortunately I don't think we have an existing example in policy-library, but I found one example on stackoverflow: https://stackoverflow.com/questions/49822794/kubernetes-custom-resource-definition-required-field

Reply:
This example helps, but I am still not sure of the exact syntax for this use case, since there is not a fixed name for the objects in that list, its pretty much a list of objects like:

`label_key: label_value_pattern
`

so it's a dynamic field... I wil do some research on this.